### DISCUSSION TITLE
Adding uploads / media as inline in Lexical Editor

### DISCUSSION THREAD

**Discussion by tyteen4a03** [payloadcms/payload#5072] [Roadmap]
This doesn't seem to be possible right now; it would be great if we can add any type of media (video or image, external or upload) inline.

**Comment by Arinji2** [comment]
Yup, it's a really needed feature for most documentations, since lists do have images.

If the feature can't be added now atleast.. is there any workaround for this other than making a custom block

**Comment by tsemachh** [comment]
Just for curiosity all the features we see in [lexical playgorund](https://playground.lexical.dev/)
Can they imported easily to payload ? or this depends ?
I am asking because it has the inline image in the insert menu.

BTW: for now I just added to custom blocks RichTextImageLeft and Right so it will allow side by side rich text but it's less intuitive and the text doesn't flows as inline

**Comment by DanRibbens** [comment]
I moved this to the roadmap, high priority as it is a definite need to have.

Thanks!

**Comment by jmikrut** [comment]
Just a note - this is now possible with Lexical's `inlineBlocks` functionality!
